pysoa/common/transport/redis_gateway/backend/sentinel.py

```diff
@@ -117,7 +135,18 @@ def _get_connection(self, index=None):
                 )
             )
 
-        return self._master_for(self._services[index])
+        for i in range(-1, self._sentinel_failover_retries):
```

Comment:
It's nicer to write this as `for i in range(self._sentinel_failover_retries + 1)` IMO.